src/ebonite/api/images.py

```diff
@@ -4,76 +4,174 @@
 from flask import Blueprint, Response, jsonify, request
 from pyjackson.pydantic_ext import PyjacksonModel
 
-from ebonite.api.helpers import ModelIdValidator
+from ebonite.api.helpers import BuildableValidator, TaskIdValidator
 from ebonite.client.base import Ebonite
-from ebonite.core.errors import ExistingImageError, ImageWithInstancesError, NonExistingImageError
+from ebonite.core.errors import ExistingImageError, ImageWithInstancesError
 from ebonite.core.objects import Image
 
 
-class BuildImageBody(PyjacksonModel):
-    __type__ = Image
-    __include__ = ['name', 'model_id']
-    __force_required__ = ['model_id']
-
-
 class UpdateImageBody(PyjacksonModel):
     __type__ = Image
-    __include__ = ['id', 'name', 'model_id']
-    __force_required__ = ['id', 'model_id']
+    __include__ = ['id', 'name', 'task_id']
+    __force_required__ = ['id', 'task_id']
 
 
 def images_blueprint(ebonite: Ebonite) -> Blueprint:
     blueprint = Blueprint('images', __name__, url_prefix='/images')
 
     @blueprint.route('', methods=['GET'])
     def get_images() -> Tuple[Response, int]:
-        model_id = request.args.get('model_id')
-        ModelIdValidator(model_id=model_id)
-        model = ebonite.meta_repo.get_model_by_id(model_id)
-        if model is not None:
-            return jsonify([ebonite.meta_repo.get_image_by_id(x) for x in model.images]), 200
+        """
+        Gets images from metadata repository for given task
+        ---
+        parameters:
+          - name: task_id
+            in: query
+            type: integer
+            required: true
+        responses:
+          200:
+            description: List of images belonging to the project
+          404:
+            description: Task with given id does not exist
+        """
+        task_id = request.args.get('task_id')
+        TaskIdValidator(task_id=task_id)
+        task = ebonite.meta_repo.get_task_by_id(task_id)
+        if task is not None:
+            return jsonify([pj.serialize(x) for x in ebonite.get_images(task)]), 200
         else:
-            return jsonify({'errormsg': f'Model with id {model_id} does not exist'})
+            return jsonify({'errormsg': f'Task with id {task_id} does not exist'}), 404
```

Comment:
maybe it's will be more readable if you use raise specific exception here and add error handler for it